geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/GeodeRedisSetSynchronized.java

```diff
@@ -59,8 +60,19 @@ public long srem(Collection<ByteArrayWrapper> membersToRemove) {
       Set<ByteArrayWrapper> newValue = createSet(oldValue);
       newValue.removeAll(membersToRemove);
       removedCount.set(oldValue.size() - newValue.size());
+
+      // if(newValue.isEmpty()) {
+      // context.getRegionProvider().removeKey(key, context.getKeyRegistrar().getType(key));
+      // }
+
       return newValue;
     });
+
+    if (members().isEmpty()) {
+      RedisDataType type = context.getKeyRegistrar().getType(key);
```

Comment:
It seems wrong here to refetch the RedisDataType and then blow it away without checking if the type is what we expect. For example here should we check to see if it is == REDIS_SET before calling removeKey?

Reply:
We actually check the data type in the SREM executor before we create the synchronized set and call SREM.  Do you think we need to check it again?

Reply:
We added a test to make sure the error is thrown and check the message.